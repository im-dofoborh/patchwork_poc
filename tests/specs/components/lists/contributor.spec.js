import { mount, createLocalVue } from '@vue/test-utils';
import ContributorList from '~/components/lists/Contributor';

describe('ContributorList component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('contributor-list', ContributorList);
    wrapper = mount(ContributorList, {
      localVue,
    });
  })

  describe('renders', () => {
    it('default variant', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('with contributors', () => {
      wrapper.setData({
        contributors: [
          {
            id: 'foo',
            login: 'Foo',
            html_url: 'https://github.com/foo',
            avatar_url: 'https://github.com/images/foo.png',
          },
          {
            id: 'bar',
            login: 'Bar',
            html_url: 'https://github.com/bar',
            avatar_url: 'https://github.com/images/bar.png',
          }
        ]
      })
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('getUniqueAuthors', () => {
    it('returns unique authors', () => {
      const author = {
        id: 'foo',
      };
      const commits = [
        {
          author,
        },
        {
          author,
        },
      ];
      expect(wrapper.vm.getUniqueAuthors(commits)).toEqual([author]);
    });

    it('ignores commits without authors', () => {
      const commits = [
        {
          sha: '123',
        },
      ];
      expect(wrapper.vm.getUniqueAuthors(commits)).toEqual([]);
    });
  });
});
