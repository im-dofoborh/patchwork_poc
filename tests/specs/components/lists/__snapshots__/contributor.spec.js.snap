// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributorList component renders default variant 1`] = `<span name="fade" tag="ul" class="sg-contributor-list"></span>`;

exports[`ContributorList component renders with contributors 1`] = `<span name="fade" tag="ul" class="sg-contributor-list"><li class="sg-contributor-list__item"><a href="https://github.com/foo" title="Foo" target="_blank" class="sg-contributor-list__link"></a></li><li class="sg-contributor-list__item"><a href="https://github.com/bar" title="Bar" target="_blank" class="sg-contributor-list__link"></a></li></span>`;
